### PR TITLE
Feat/query sorted by time

### PR DESCRIPTION
**Pull request by Mark-Niemeyer**

### CODE REVIEW

**Comment by jarkenau** [comment]
@Mark-Niemeyer what's the status on this PR?

File: seerep_srv/seerep_core/src/core_dataset.cpp [hunk 1]
```diff
@@ -448,14 +497,29 @@ CoreDataset::getAllDatasetUuids(std::shared_ptr<seerep_core::CoreDataset::Dataty
...
   {
-    allIds.push_back(std::move(it->second));
+    std::vector<std::pair<int64_t, boost::uuids::uuid>> sortingVector;
+    for (auto it = std::make_move_iterator(allIdsFromTimeTree.begin()),
+              end = std::make_move_iterator(allIdsFromTimeTree.end());
+         it != end; ++it)
+    {
+      sortingVector.push_back(std::make_pair(boost::geometry::get<0>(it->first.min_corner()), it->second));
+    }
+    return sortResultByTimeActualSorting(sortingVector);
```
Comment (jarkenau) [inline seerep_srv/seerep_core/src/core_dataset.cpp:510]: Please reuse the code from [sortResultByTime](https://github.com/agri-gaia/seerep/blob/f0c986de31ff366c70b2b698474e7dd30d0030f6/seerep_srv/seerep_core/src/core_dataset.cpp#L427)

File: seerep_srv/seerep_core/src/core_dataset.cpp [hunk 2]
```diff
@@ -401,11 +401,59 @@ CoreDataset::intersectQueryResults(std::optional<std::vector<seerep_core_msgs::A
...
+{
+  std::sort(sortingVector.begin(), sortingVector.end());
+  std::vector<boost::uuids::uuid> sortedResults;
+  for (auto x : sortingVector)
+  {
+    std::cout << x.first << std::endl;
+    sortedResults.push_back(x.second);
+  }
+  return sortedResults;
+}
```
Comment (jarkenau) [inline seerep_srv/seerep_core/src/core_dataset.cpp:454]:
```suggestion
  std::sort(sortingVector.begin(), sortingVector.end());
  std::vector<boost::uuids::uuid> sortedResults;
  sortedResults.reserve(sortingVector.size());
  
  for (auto&  x : sortingVector)
  {
    sortedResults.push_back(std::move(x.second));
  }
  return sortedResults;
}
```

File: seerep_srv/seerep_core/include/seerep_core/core_dataset.h [hunk 3]
```diff
@@ -290,22 +290,40 @@ class CoreDataset
...
+  sortResultByTime(std::optional<std::vector<seerep_core_msgs::AabbTimeIdPair>>& timetree_result,
+                   std::set<boost::uuids::uuid> intersectionResult);
+  /**
+   * @brief does the actual sorting for the sortResultByTime method
+   *
+   * @param sortingVector the data to be sorted
+   * @return std::vector<boost::uuids::uuid> sorted vector of result uuids
+   */
+  std::vector<boost::uuids::uuid>
+  sortResultByTimeActualSorting(std::vector<std::pair<int64_t, boost::uuids::uuid>>& sortingVector);
```
Comment (jarkenau) [inline seerep_srv/seerep_core/include/seerep_core/core_dataset.h:311]: Combine these functions? I don't think it makes sense to call them individually, and the naming is not the best.

**Review by jarkenau — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 4]
```diff
@@ -59,3 +59,5 @@ wheels/
 .installed.cfg
 *.egg
 MANIFEST
+
+**/other_packages/**
```
[the remark this answers is not in the document]
Comment (Mark-Niemeyer) [reply .gitignore:63]: This allows to add other repos to the other_packages folder when these other repos link to seerep, but aren't part of this repo. This is a quick solution. We should create debian packages or install this repo properly in the long run.

**Review by jarkenau — APPROVED** [review]

**Pull request merged**